awscli/customizations/ecs/deploy.py

```diff
@@ -341,6 +342,24 @@ def describe_cd_resources(self):
             raise exceptions.ServiceClientError(
                 action='describe Code Deploy deployment group', error=e)
 
+        try:
+            dgp_info = self.deployment_group_details['deploymentGroupInfo']
+            deployment_config_name = dgp_info['deploymentConfigName']
+            self.deployment_config = self._client.get_deployment_config(
+                deploymentConfigName=deployment_config_name)
+        except ClientError as e:
+            raise exceptions.ServiceClientError(
+                action='describe Code Deploy deployment config', error=e)
+
+    def get_deployment_duration(self):
+        wait_time = self.get_deployment_wait_time()
+        rerouting_time = self.get_traffic_rerouting_time()
+
+        if wait_time is None or rerouting_time is None:
```

Comment:
To be honest I do not understand why `get_deployment_wait_time` returns `None` in the case of no information instead of just 0. I would propose to return 0 from `get_deployment_wait_time` and `get_traffic_rerouting_time` instead of the `None` special use case. WDYT?